inference-engine/src/transformations/src/transformations/optimize_strided_slice.cpp

```diff
@@ -18,7 +18,16 @@ bool ngraph::pass::UselessStridedSliceEraser::run_on_function(std::shared_ptr<ng
             continue;
         if (ss->input(0).get_shape() != ss->output(0).get_shape())
             continue;
-        rewritten |= replace_output_update_name(ss->output(0), ss->input_value(0));
+
+        bool is_reverse = false;
+        auto strides = std::dynamic_pointer_cast<ngraph::op::Constant>(ss->input_value(3).get_node_shared_ptr())->cast_vector<int64_t>();
```

Comment:
1. Use `opset3::Constant`
2. You need to check that dynamic cast returned not nullptr
3. Use std::any_of to check that at least one of values less than 0

Reply:
fixed